Update CHANGELOG.md for 0.7.1 (#361)
mobilecoinfoundation/sgx: CHANGELOG.md

## [Unreleased] - ReleaseDate

## [0.7.1] - 2023-07-12

### Added

- Added `FromHex` implementation for MrEnclave and MrSigner

> - Added `From<quote3<&[u8]>> for Quote<Vec<u8>>`

Reviewer: Especially since we were talking past each other initially should use the correct type names

```suggestion
- Added `From<Quote3<&[u8]>> for Quote3<Vec<u8>>`
```

Author: My bad, I must've missed the Shift key

## [Unreleased] - ReleaseDate

## [0.7.1] - 2023-07-12

### Added

- Added `FromHex` implementation for MrEnclave and MrSigner
- Added `From<quote3<&[u8]>> for Quote<Vec<u8>>`

### Fixed

- Fixed `ExtendedFeatureRequestMask::AMX`

>  - Previously the `ExtendedFeatureRequestMask::AMX` was mapped to `SGX_XFRM_LEGACY`. Now `ExtendedFeatureRequestMask::AMX` correctly maps to `SGX_XFRM_AMX`.

Reviewer: ```suggestion
- Fixed `ExtendedFeatureRequestMask::AMX`. Previously it was mapped to `SGX_XFRM_LEGACY`. Now it correctly maps to `SGX_XFRM_AMX`.
```

Reviewer: I must have fatfingered cancel review, resolving this, it no longer applies

>     - Previously the `ExtendedFeatureRequestMask::AMX` was mapped to `SGX_XFRM_LEGACY`. Now `ExtendedFeatureRequestMask::AMX` correctly maps to `SGX_XFRM_AMX`.

Reviewer: ```suggestion
- Fixed `ExtendedFeatureRequestMask::AMX`. Previously the `ExtendedFeatureRequestMask::AMX` was mapped to `SGX_XFRM_LEGACY`. Now `ExtendedFeatureRequestMask::AMX` correctly maps to `SGX_XFRM_AMX`.
```
